packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx

```diff
@@ -116,11 +116,9 @@ function Example(props) {
   return (
     <span style={{position: 'relative'}}>
       <button ref={ref} {...triggerProps}>I have a tooltip</button>
-      {state.isOpen && (
-        <Tooltip state={state} {...tooltipProps}>
-          And the tooltip tells you more information.
-        </Tooltip>
-      )}
+      <Tooltip hidden={!state.isOpen} state={state} {...tooltipProps}>
```

Comment:
I think the difference for anyone who copied from the prior example would be that the `aria-describedby` prop on the trigger referencing the Tooltip element will persist even when the Tooltip is not in the DOM. This minor failure would be imperceivable to assistive technology users, but might trigger warnings with automated accessibility testing.

Reply:
btw, hidden isn't actually needed, it's in state if it's needed inside the component

Reply:
> > In my tests with NVDA and JAWS, the Tooltip is not immediately announced when using the virtual cursor, so perhaps we decided that persisting the Tooltip with aria-describedby was a moot point.
> 
> yep! that was it!

With JAWS, INSERT+TAB or CAPS LOCK+TAB (on laptop) announces the button and tooltip for the button focused with the virtual cursor.

With NVDA INSERT+numpad5 or CAPS LOCK+SHIFT+O (on laptop) announces the button and tooltip for the button focused in browse mode. [hidden] attribute makes no difference.

So, in both cases there is a way to access the tooltip info for an object that would not be available if the Tooltip and `aria-describedby` attribute does not persist in the DOM.

Reply:
> btw, hidden isn't actually needed, it's in state if it's needed inside the component

 `[hidden]` is necessary, because the `tooltipProps` returned by `useTooltip` do not return anything related to the `isOpen` state: https://github.com/adobe/react-spectrum/blob/622e33dd0b8c468463fe47a385a75e3138dee71c/packages/%40react-aria/tooltip/src/useTooltip.ts#L37-L41

Reply:
What I mean is that it should be in the aria hook and should be returned from useTooltip

Reply:
An alternative example adding a CSS animation to set `opacity` and `visibility` based on `isOpen` state:

```tsx 
import {useTooltipTriggerState} from '@react-stately/tooltip';
import {mergeProps} from '@react-aria/utils';

function Tooltip({state, ...props}) {
  let {tooltipProps} = useTooltip(props, state);
  let {isOpen} = state;

  return (
    <span
      style={{
        position: 'absolute',
        left: '5px',
        top: '100%',
        marginTop: '10px',
        backgroundColor: 'white',
        color: 'black',
        padding: '5px',
        /* fade in or fade out tooltip based on isOpen state */
        opacity: (isOpen ? 1 : 0),
        visibility: (isOpen ? 'visible' : 'hidden'),
        transition: `
          opacity 130ms ease-in-out 0ms,
          visibility 0ms linear ${isOpen ? 0 : 130}ms
        `
      }}
      {...mergeProps(props, tooltipProps)} >
      {props.children}
    </span>
  );
}

function Example(props) {
  let state = useTooltipTriggerState(props);
  let ref = React.useRef();

  // Get props for the trigger and its tooltip
  let {triggerProps, tooltipProps} = useTooltipTrigger(props, state, ref);

  return (
    <span style={{position: 'relative'}}>
      <button ref={ref} {...triggerProps}>I have a tooltip</button>
      <Tooltip state={state} {...tooltipProps}>
        And the tooltip tells you more information.
      </Tooltip>
    </span>
  );
}

<Example />
<Example />
```

Reply:
Ok, I see, we technically have two tooltips... https://github.com/adobe/react-spectrum/pull/2102/files#diff-225c01ce8d5eeafee3ecca35c4f8cded18543869468adbfb9b956914bde5280bR74 and they are in different places
whereas this mdx file only has one

could we wrap the new hidden tooltip into a new component (HiddenTooltip) that calls useTooltip which would return hidden when state is closed? I don't think it'll matter if the hidden element has role=tooltip? otherwise we can remove the role when hidden. This way we don't have to move the overlay. It should also fix this example so the user only has to remove the `{isOpen &&`

the aria hooks should take care of everything for the user

Reply:
Ok I added a note to the description before the example about why `hidden` is used here.